src/main/java/bio/terra/common/exception/SamApiException.java

```diff
@@ -0,0 +1,85 @@
+package bio.terra.common.exception;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.client.http.HttpStatusCodes;
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsde.workbench.client.sam.ApiException;
+import org.broadinstitute.dsde.workbench.client.sam.model.ErrorReport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+
+/**
+ * Wrapper exception for non-200 responses from calls to Sam. Does not extend ErrorReportException
+ * so that SamApiException is a checked exception type. Can be converted to an ErrorReportException
+ * subtype using the toErrorReportException function.
+ */
+public class SamApiException extends Exception {
+  private ApiException apiException;
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+  private static Logger logger = LoggerFactory.getLogger(SamApiException.class);
+
+  public static SamApiException createSamApiException(ApiException apiException) {
```

Comment:
nit: `create` is a fine name for this. The `SamApiException` is implicit in the class name for the factory function. e.g. `SamApiException.createSamApiException` is repetitive.

Reply:
Done

Reply:
It does not look like you changed the method name.

Reply:
Shows up for me as changed now...